batching.go

```diff
@@ -9,27 +9,45 @@ import (
 )
 
 type batch struct {
-	ds    *Datastore
-	batch *pgx.Batch
+	ds           *Datastore
+	batch        *pgx.Batch
+	maxBatchSize uint16
 }
 
 // Batch creates a set of deferred updates to the database.
 func (d *Datastore) Batch() (ds.Batch, error) {
-	return &batch{ds: d, batch: &pgx.Batch{}}, nil
+	b := &batch{ds: d, batch: &pgx.Batch{}, maxBatchSize: 0}
+	b.batch.Queue("BEGIN")
+	return b, nil
+}
+
+// Set the max batch size (0 or default means unlimited - the batch is only cleared when calling Commit)
+func (b *batch) SetMaxBatchSize(size uint16) {
+	b.maxBatchSize = size
+}
+
+func (b *batch) checkMaxBatchSize() error {
```

Comment:
"check" implies something passive, but this commits, so maybe we could call it `commitWhenFullBatch` or something like that?

Reply:
good call